osquery/carver/carver.h

```diff
@@ -0,0 +1,121 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <osquery/dispatcher.h>
+#include <osquery/filesystem.h>
+
+namespace osquery {
+
+class Carver : public InternalRunnable {
+ public:
+  Carver(const std::set<std::string>& paths, const std::string& guid);
+
+  ~Carver();
+
+  /*
+   * @brief A helper function to perform a start to finish carve
+   *
+   * This function walks through the carve, compress, and exfil functions
+   * in one fell swoop. Use of this class should largely happen through
+   * this function.
+   */
+  void start();
+
+ protected:
```

Comment:
If nothing inherits from carver (or we plan for nothing to inherit from it) we can just make these private.